ksql-rest-app/src/test/java/io/confluent/ksql/rest/client/KsqlRestClientTest.java

```diff
@@ -76,11 +77,71 @@ public void testKsqlResource() {
 
 
   @Test
-  public void testStreamQuery() {
+  public void testStreamQuery() throws InterruptedException {
+    MockStreamedQueryResource sqr = mockApplication.getStreamedQueryResource();
     RestResponse<KsqlRestClient.QueryStream> queryResponse = ksqlRestClient.makeQueryRequest
         ("Select *");
     Assert.assertNotNull(queryResponse);
     Assert.assertTrue(queryResponse.isSuccessful());
+    List<MockStreamedQueryResource.TestStreamWriter> writers = sqr.getWriters();
+    Assert.assertEquals(1, writers.size());
+    MockStreamedQueryResource.TestStreamWriter writer = writers.get(0);
+    writer.enq("hello");
+    KsqlRestClient.QueryStream queryStream = queryResponse.getResponse();
+    Thread t = new Thread() {
+      public void run() {
+        queryStream.hasNext();
+     }
+    };
+    t.start();
+    t.join(10000);
+    Assert.assertFalse(t.isAlive());
+    Assert.assertTrue(queryStream.hasNext());
+    StreamedRow sr = queryStream.next();
+    Assert.assertNotNull(sr);
+    GenericRow row = sr.getRow();
+    Assert.assertEquals(1, row.getColumns().
+            size());
+    Assert.assertEquals("hello", row.getColumns().
+            get(0));
+    writer.finished();
+  }
+
+  @Test
+  public void testInterruptStreamedQuery() throws InterruptedException {
+    MockStreamedQueryResource sqr = mockApplication.getStreamedQueryResource();
```

Comment:
The first part of this test looks identical to the first part of the test above, i.e, same assertions. No need to assert the same stuff again as it has been proven to work already.